src/sdl/utils.cpp

```diff
@@ -769,24 +771,20 @@ bool in_mask_surface(const surface& nsurf, const surface& nmask)
 		return false;
 	}
 
-	{
-		const_surface_lock lock(nsurf);
-		const_surface_lock mlock(nmask);
+	const_surface_lock lock(nsurf);
+	const_surface_lock mlock(nmask);
 
-		const uint32_t* mbeg = mlock.pixels();
-		const uint32_t* mend = mbeg + nmask->w*nmask->h;
-		const uint32_t* beg = lock.pixels();
-		// no need for 'end', because both surfaces have same size
-
-		while(mbeg != mend) {
-			uint8_t malpha = (*mbeg) >> 24;
-			if(malpha == 0) {
-				uint8_t alpha = (*beg) >> 24;
-				if (alpha)
-					return false;
-			}
-			++mbeg;
-			++beg;
+	utils::span surf_pixels = lock.pixel_span();
+	utils::span mask_pixels = mlock.pixel_span();
+
+	// Note: unlike in mask_surface, both ranges here have the same size.
+	for(std::size_t i = 0; i < surf_pixels.size(); ++i) {
+		const uint32_t surf_alpha = surf_pixels[i] & SDL_ALPHA_MASK;
+		const uint32_t mask_alpha = mask_pixels[i] & SDL_ALPHA_MASK;
+
+		// A non-transparent surface pixel corresponding to a transparent mask pixel.
+		if(surf_alpha && mask_alpha == SDL_ALPHA_TRANSPARENT) {
```

Comment:
`SDL_ALPHA_TRANSPARENT` makes the code less readable, because it means "a fully masked pixel" rather than "a pixel which would be visible through the mask".